File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/service/broadcasters/LocalPlaybackBroadcaster.java
package com.lasthopesoftware.bluewater.client.playback.service.broadcasters;

import android.content.Intent;
import android.support.v4.content.LocalBroadcastManager;
import com.lasthopesoftware.bluewater.client.playback.file.PositionedFile;

public class LocalPlaybackBroadcaster implements IPlaybackBroadcaster {
    private final LocalBroadcastManager localBroadcastManager;

    public LocalPlaybackBroadcaster(LocalBroadcastManager localBroadcastManager) {
        this.localBroadcastManager = localBroadcastManager;
    }

    @Override
    public void sendPlaybackBroadcast(final String broadcastMessage, final int libraryId, final PositionedFile positionedFile) {
        final Intent playbackBroadcastIntent = new Intent(broadcastMessage);

        final int currentPlaylistPosition = positionedFile.getPlaylistPosition();

        final int fileKey = positionedFile.getServiceFile().getKey();

        playbackBroadcastIntent
			.putExtra(PlaylistEvents.PlaylistParameters.playlistPosition, currentPlaylistPosition)
			.putExtra(PlaylistEvents.PlaybackFileParameters.fileLibraryId, libraryId)
			.putExtra(PlaylistEvents.PlaybackFileParameters.fileKey, fileKey);

        localBroadcastManager.sendBroadcast(playbackBroadcastIntent);
    }
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/service/broadcasters/LocalPlaybackBroadcaster.kt
package com.lasthopesoftware.bluewater.client.playback.service.broadcasters

import android.content.Intent
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.lasthopesoftware.bluewater.client.playback.file.PositionedFile
import com.lasthopesoftware.bluewater.shared.android.messages.SendMessages

class LocalPlaybackBroadcaster(private val sendMessages: SendMessages) : IPlaybackBroadcaster {
	override fun sendPlaybackBroadcast(broadcastMessage: String, libraryId: LibraryId, positionedFile: PositionedFile) {
		val playbackBroadcastIntent = Intent(broadcastMessage)
		val currentPlaylistPosition = positionedFile.playlistPosition
		val fileKey = positionedFile.serviceFile.key
		playbackBroadcastIntent
			.putExtra(PlaylistEvents.PlaylistParameters.playlistPosition, currentPlaylistPosition)
			.putExtra(PlaylistEvents.PlaybackFileParameters.fileLibraryId, libraryId.id)
			.putExtra(PlaylistEvents.PlaybackFileParameters.fileKey, fileKey)
		sendMessages.sendBroadcast(playbackBroadcastIntent)
	}
}
